concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -512,112 +534,110 @@ exportSections dbDir outDir chunkSize genIndex startHeight blockIndex lastWritte
-                                                                (_bpHeight . sbInfo $ sb)
-                                                                outDir
-                                                                chunkSize
-                                                                lastWrittenChunkM
-                                                                -- An initial value for the @FinalizationIndex@. This will be updated
-                                                                -- as we export blocks with a finalization index associated.
-                                                                -- The finalization record itself is only exported if the block it
-                                                                -- points to is the last block of the section, hence 0 is a reasonable
-                                                                -- value for starting the export.
-                                                                0
-                                                                getBlockAt
-                                                                getFinalizationRecordAt
-                                                        return (False, singleton (genHash, chunks))
+                            Just (KonsensusV1.VersionDatabaseHandlers (dbh :: KonsensusV1.DatabaseHandlers pv)) ->
+                                runReaderT
+                                    ( KonsensusV1.runDiskLLDBM $ do
+                                        exportResult <- exportConsensusV1Blocks @pv outDir chunkSize genIndex startHeight blockIndex lastWrittenChunkM
+                                        liftIO $ KonsensusV1.closeDatabase dbh
+                                        return exportResult
                                     )
-                                    (DBState dbh)
-                    liftIO $ closeDatabase dbh
-                    -- if an error occurred, return sections
-                    -- that were successfully written to the
-                    -- file system; otherwise export the section
-                    -- corresponding to the incremented genesis
-                    -- index.
-                    if exportError
-                        then return (True, sectionData)
-                        else do
-                            (err, secs) <-
-                                exportSections
-                                    dbDir
-                                    outDir
-                                    chunkSize
-                                    (genIndex + 1)
-                                    1
-                                    Empty
-                                    Nothing
-                            return (err, sectionData >< secs)
+                                    dbh
+                    Just (VersionDatabaseHandlers (dbh :: DatabaseHandlers pv ())) -> do
+                        exportResult <- do
```

Comment:
If possible, can we refactor `exportSections` a bit to reduce the excessive indentation?